### ISSUE TITLE
Some questions about the data part

### ISSUE THREAD

**Issue by chenchongthu** [staoxiao/RetroMAE#11]
Great work, I have some questions below:

class RetroMAECollator(DataCollatorForWholeWordMask):
    max_seq_length: int = 512
    encoder_mlm_probability: float = 0.15
    decoder_mlm_probability: float = 0.15

    def __call__(self, examples):
        input_ids_batch = []
        attention_mask_batch = []
        encoder_mlm_mask_batch = []
        decoder_labels_batch = []
        decoder_matrix_attention_mask_batch = []

        tgt_len = self.max_seq_length - self.tokenizer.num_special_tokens_to_add(False)

        for e in examples:
            e_trunc = self.tokenizer.build_inputs_with_special_tokens(e['token_ids'][:tgt_len])
            tokens = [self.tokenizer._convert_id_to_token(tid) for tid in e_trunc]

            self.mlm_probability = self.encoder_mlm_probability
            text_encoder_mlm_mask = self._whole_word_mask(tokens)

            self.mlm_probability = self.decoder_mlm_probability
            mask_set = []
            for _ in range(min(len(tokens), 256)):
                mask_set.append(self._whole_word_mask(tokens))

First, what does 256 mean here, why is it set to 256 ?

Second, e['token_ids'] has several blocks according to process.py.   It seems that e['token_ids'] is  list[list[int]], so (e['token_ids'][:tgt_len] seems not right?

**Comment by staoxiao** [comment]
Thanks for your interest in RetroMAE!

For the enhanced decoder, the collater needs to prepare a mask list for each token. If the number of tokens is large, the data processing may be time-consuming. Therefore, I set a mask set with a limited max size of 256 (maybe another hyper-parameter is better),  and then mask list will be sampled from this set.  

Blocks returned by pad_each_line function will be assigned to multiple rows ({'token_ids': list[list[int]]} will be [{'token_ids':list[int]}, ..]), so the e['token_ids'] is list[int].
